src/torchmetrics/image/uqi.py

```diff
@@ -82,29 +84,43 @@ def __init__(
         **kwargs: Any,
     ) -> None:
         super().__init__(**kwargs)
-        rank_zero_warn(
-            "Metric `UniversalImageQualityIndex` will save all targets and"
-            " predictions in buffer. For large datasets this may lead"
-            " to large memory footprint."
-        )
-
-        self.add_state("preds", default=[], dist_reduce_fx="cat")
-        self.add_state("target", default=[], dist_reduce_fx="cat")
+        if reduction not in ("elementwise_mean", "sum", "none", None):
+            raise ValueError(
+                f"The `reduction` {reduction} is not valid. Valid options are `elementwise_mean`, `sum`, `none`, None."
+            )
+        if reduction == "none" or reduction is None:
+            rank_zero_warn(
+                "Metric `UniversalImageQualityIndex` will save all targets and predictions in the buffer when using"
+                "`reduction=None` or `reduction='none'. For large datasets, this may lead to a large memory footprint."
+            )
+            self.add_state("preds", default=[], dist_reduce_fx="cat")
+            self.add_state("target", default=[], dist_reduce_fx="cat")
+        else:
+            self.add_state("sum_uqi", tensor(0.0), dist_reduce_fx="sum")
+            self.add_state("numel", tensor(0), dist_reduce_fx="sum")
         self.kernel_size = kernel_size
         self.sigma = sigma
         self.reduction = reduction
 
     def update(self, preds: Tensor, target: Tensor) -> None:
         """Update state with predictions and targets."""
         preds, target = _uqi_update(preds, target)
-        self.preds.append(preds)
-        self.target.append(target)
+        if self.reduction == "none" or self.reduction is None:
```

Comment:
```suggestion
        if self.reduction is None or self.reduction == "none":
```
I think that `None == "none"` is into valid